generator/internal/idempotency_policy_generator.cc

```diff
@@ -155,7 +155,25 @@ Status IdempotencyPolicyGenerator::GenerateCc() {
     "  }\n");
   // clang-format on
 
+  auto is_set_iam_policy = [](google::protobuf::MethodDescriptor const& m) {
+    return m.output_type()->full_name() == "google.iam.v1.Policy" &&
+           m.input_type()->full_name() == "google.iam.v1.SetIamPolicyRequest";
+  };
```

Comment:
Q. Perhaps this deserves a `IsSetIamPolicy()` helper in `predicate_utils.h`?

We do have some similar logic in `ClientGenerator::ClientGenerator()`, but it isn't as simple.

Reply:
The `ClientGenerator()` is trying to do something more interesting, the signature extensions are needed to use the `SetIamPolicy(resource, policy)` and `GetIamPolicy(resource)` overloads, and to use them in the loop for `SetIamPolicy(resource, updater)`.

 By the time a request gets to the idempotency policy the request is always in the form of a request protos and we do not need the additional complexity.

I am going to leave this as-is.